field.go

```diff
@@ -1172,10 +1200,23 @@ func (f *Field) importValue(columnIDs []uint64, values []int64, options *ImportO
 	dataByFragment := make(map[importKey]importValueData)
 	for i := range columnIDs {
 		columnID, value := columnIDs[i], values[i]
-		if value > bsig.Max {
-			return fmt.Errorf("%v, columnID=%v, value=%v", ErrBSIGroupValueTooHigh, columnID, value)
-		} else if value < bsig.Min {
-			return fmt.Errorf("%v, columnID=%v, value=%v", ErrBSIGroupValueTooLow, columnID, value)
+
+		// Increase the maximum BSI value, if necessary.
+		uvalue := uint64(value)
+		if value < 0 {
+			uvalue = uint64(-value)
+		}
+
+		if int64(uvalue) > bsig.Max {
+			if err := func() error {
+				f.mu.Lock()
+				defer f.mu.Unlock()
+				bsig.Max = int64(uvalue)
+				f.options.Max = int64(uvalue)
```

Comment:
I think there is a race here if multiple imports are going on at once. Because f.options.Max might be changed concurrently, and we don't have a read lock on it since we're effectively caching the value with bsig.Max. Another goroutine might set it higher, and then we come along and set it higher than our previous known value, but lower than what the other routine just changed it to.